src/yb/util/memory/arena.cc

```diff
@@ -253,13 +255,17 @@ void ArenaBase<Traits>::Reset(ResetMode mode) {
     // detection of memory-related bugs (invalid shallow copies, etc.).
     size_t last_size = current->full_size();
     current->Destroy(buffer_allocator_);
+    UntrackMemory(arena_footprint_);
     arena_footprint_ = 0;
     ReleaseStoreCurrent(nullptr);
     AddComponentUnlocked(NewBuffer(last_size, 0));
 #else
+    size_t bytes_before = arena_footprint_;
     arena_footprint_ = current->full_size();
+    TrackMemoryDifference(bytes_before, arena_footprint_);
 #endif
   } else {
+    UntrackMemory(arena_footprint_);
```

Comment:
Btw, we should also release `arena_footprint_` memory at destructor `ArenaBase<Traits>::~ArenaBase()`.

Reply:
I am a little confused here. Is that what is being done?

Reply:
In `ArenaBase<Traits>::~ArenaBase()` destructor, `AcquireLoadCurrent()->Destroy(buffer_allocator_);` deallocates all components(inside of this function, it usea while loop to delete all nodes from the linkedlist), so we should call `UntrackMemory(arena_footprint_);` in the destructor.